src/store/assets/tokens.ts

```diff
@@ -59,9 +60,13 @@ const assetItemState = atomFamily<Asset, AssetWithMeta>(asset => {
   return anAtom;
 }, deepEqual);
 
-export const assetsState = atom(get =>
-  get(waitForAll(transformAssets(get(currentAccountBalancesState)).map(assetItemState)))
-);
+export const assetsState = atom(get => {
+  get(currentNetworkState);
+  const balances = get(currentAccountBalancesState);
+  const transformed = transformAssets(balances);
+  const arr = transformed.map(assetItemState).map(anAtom => get(anAtom));
```

Comment:
Can you explain for me what is happening on this line?

Reply:
sure! [This is the most updated version](https://github.com/blockstack/stacks-wallet-web/blob/7f257df29fa8d9ea7d41b0c886c49e01dda5fdd9/src/store/assets/tokens.ts#L62-L81), this comment is on an outdated version. 

```ts
export const baseAssetsAnchoredState = atom(get => {
  const balances = get(currentAnchoredAccountBalancesState);
  return transformAssets(balances);
});

export const assetsAnchoredState = atom(get => {
  const network = get(currentNetworkState);
  const transformed = get(baseAssetsAnchoredState);
  return get(waitForAll(transformed.map(asset => assetItemState([asset, network.url]))));
});

export const baseAssetsUnanchoredState = atom(get => {
  const balances = get(currentAccountBalancesUnanchoredState);
  return transformAssets(balances);
});

export const assetsUnanchoredState = atom(get => {
  const network = get(currentNetworkState);
  const transformed = get(baseAssetsUnanchoredState);
  return get(waitForAll(transformed.map(asset => assetItemState([asset, network.url]))));
});
```
I've updated it so we can access the base assets array (what comes from `transformAssets`) while we wait for the assetsState atom to load.

Generally it works like this:
- get the balances
- `transformAssets` takes the object and returns an array of `Asset` type
- `assetsUnanchoredState` and `assetsAnchoredState` then use those arrays to fetch meta data for tokens that require them